src/core/surface/completion_queue.c

```diff
@@ -94,7 +94,7 @@ static void on_pollset_shutdown_done(grpc_exec_ctx *exec_ctx, void *cc,
 void grpc_cq_global_init(void) { gpr_mu_init(&g_freelist_mu); }
 
 void grpc_cq_global_shutdown(void) {
-  gpr_mu_destroy(&g_freelist_mu);
```

Comment:
This doesn't seem like a fix at all: if there's a thread that's able to manipulate the freelist here, then it's still able to manipulate the freelist after the unlock.

Suggest instead doing something similar to iomgr and waiting for all completion queues to be destroyed before proceeding with shutdown.